src/Topologies/dtopology2d.jl

```diff
@@ -85,6 +86,41 @@ struct DistributedTopology2D{
     boundaries::BF
 end
 
+"""
+    spacefillingcurve(mesh::Meshes.AbstractCubedSphere)
+
+Generate element ordering, `elemorder`, based on a space filling curve
+for a `CubedSphere` mesh.
+
+"""
+function spacefillingcurve(mesh::Meshes.AbstractCubedSphere)
+    ne = mesh.ne
+    majordim = [1, 2, 1, 2, 1, 2]
+    elemorder = CartesianIndex{3}[]
+    for j in 1:6
+        push!(
+            elemorder,
+            [
+                CartesianIndex(i, j) for
+                i in gilbertindices((ne, ne); majdim = majordim[j])
+            ]...,
+        )
+    end
```

Comment:
Use slightly more descriptive names, and `append!` rather than splatting into `push!`
```suggestion
    for panel in 1:6
        append!(
            elemorder,
            [
                CartesianIndex(cartidx, panel) for
                cartidx in gilbertindices((ne, ne); majdim = majordim[panel])
            ],
        )
    end
```

Reply:
append!  is not working correctly with CartesianIndices here! All other revisions are incorporated